bindings/python/iota_sdk/types/common.py

```diff
@@ -31,8 +31,15 @@ def custom_to_dict(self, *args, **kwargs):
         # pylint: disable=protected-access
         original_dict = to_dict(self, *args, **kwargs)
 
-        result = {k: v for k, v in original_dict.items() if v is not None}
-        return result
+        # recursive remove the None values
+        def filter_none(value):
+            if isinstance(value, dict):
+                return {k: filter_none(v) for k, v in value.items() if v is not None}
+            if isinstance(value, list):
+                return [filter_none(item) for item in value if item is not None]
+            return value
+
+        return filter_none(original_dict)
```

Comment:
out of interest: was this necessary due to the single account, or just fixing a bug?

Reply:
this was needed to filter out none from any sub/recursive class with @json. before it only did the depth=1 json fields

Reply:
yeah, but wouldn't that not also be necessary indepently of this PR, like ... is it a fix, or did it just now become necessary, or do I not get the point? :joy: